src/java.base/share/classes/java/time/InstantSource.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.time;
+
+import java.time.Clock.SourceClock;
+import java.time.Clock.SystemInstantSource;
+import java.util.Objects;
+
+/**
+ * Provides access to the current instant.
+ * <p>
+ * Instances of this interface are used to find the current instant.
+ * As such, an {@code InstantSource} can be used instead of {@link System#currentTimeMillis()}.
+ * <p>
```

Comment:
The word 'current' is likely to misleading here.  The specification of an interface does not have any context in which to describe what the instant represents or what it is relative to.
Given the intended use cases, it is definitely not always related to System.currentTimeMillis() 
which is bound to the running system.
i think the best you could say is that it returns an instant provided by the source as does the 'instance()' method.

Reply:
This is the definition used by `Clock` since Java 8. It is also the purpose of the interface. ie. this isn't an interface for providing instants in general, but for providing the _current instant_. I can clarify wrt the meaning of "current", but I don't see how that word can be avoided.